### PR TITLE
Add extension placeholder for serialized medias

### PR DESCRIPTION
**Pull request by Prokyonn**

### CODE REVIEW

File: Content/Serializer/MediaSerializer.php [hunk 1]
```diff
@@ -94,15 +94,19 @@ public function serialize(MediaInterface $media, string $locale, ?SerializationC
         $preferredExtension = $this->imageConverter->getSupportedOutputImageFormats($formatMediaApi->getMimeType())[0] ?? null;
         if ($preferredExtension) {
             $fileName = \pathinfo($fileName)['filename'] . '.' . $preferredExtension;
+            $mediaData['preferredExtension'] = $preferredExtension;
```
Comment (alexander-schranz) [inline Content/Serializer/MediaSerializer.php]:
```suggestion
            $mediaData['formatPreferredExtension'] = $preferredExtension;
```

So they are beside each other in the json response.

**Comment by Prokyonn** [comment]
@alexander-schranz @wachterjohannes  to prevent the BC break, we could maybe add a config like this:

```yaml
sulu_headless:
    media:
        use_preferred_extension: true
```

WDYT?

File: Content/Serializer/MediaSerializer.php [hunk 2]
```diff
@@ -94,15 +94,19 @@ public function serialize(MediaInterface $media, string $locale, ?SerializationC
         $preferredExtension = $this->imageConverter->getSupportedOutputImageFormats($formatMediaApi->getMimeType())[0] ?? null;
         if ($preferredExtension) {
```
Comment (alexander-schranz) [inline Content/Serializer/MediaSerializer.php:95]: I think `formatUri`, `formatPreferredExtension` should be `null` if there is no preferredExtension as I think that indicates that there are no formats available for this media. E.g. `.doc`, `.xls`, ...

File: Content/Serializer/MediaSerializer.php [hunk 3]
```diff
@@ -94,15 +94,19 @@ public function serialize(MediaInterface $media, string $locale, ?SerializationC
...
             $fileName = \pathinfo($fileName)['filename'] . '.' . $preferredExtension;
+            $mediaData['preferredExtension'] = $preferredExtension;
         }
 
-        $mediaData['formatUri'] = $this->formatCache->getMediaUrl(
+        // extension brackets cannot be added here because of the urlencoding
+        $fileName = \pathinfo($fileName)['filename'] . '.' . 'extension';
```
Comment (alexander-schranz) [inline Content/Serializer/MediaSerializer.php]:
```suggestion
        $fileName = \pathinfo($fileName)['filename'] . '.' . '{extension}';
```

File: Tests/Functional/Controller/responses/navigation__get_excerpt.json [hunk 4]
```diff
@@ -80,6 +80,7 @@
                             "version": 1,
                             "subVersion": 0,
                             "name": "test-image.png",
+                            "preferredExtension": "png",
```
Comment (alexander-schranz) [inline Tests/Functional/Controller/responses/navigation__get_excerpt.json]: would replace `@string@` in `formatUri` to check the Uri is as expected.

File: Tests/Unit/Content/Serializer/MediaSerializerTest.php [hunk 5]
```diff
@@ -114,8 +114,8 @@ public function testSerialize(): void
             ->willReturn(['jpg'])
             ->shouldBeCalled();
 
-        $this->formatCache->getMediaUrl(1, 'media-1.jpg', '{format}', 1, 0)
-            ->willReturn('/media/1/{format}/media-1.jpg?v=1-0')
+        $this->formatCache->getMediaUrl(1, 'media-1.extension', '{format}', 1, 0)
+            ->willReturn('/media/1/{format}/media-1.extension?v=1-0')
```
Comment (alexander-schranz) [inline Tests/Unit/Content/Serializer/MediaSerializerTest.php]: Can we create a real instance of `LocalFormatCache` so we avoid this mock.

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

**Comment by alexander-schranz** [comment]
@Prokyonn if somebody is using `^0.9` they will not get automatically `0.10`, as `^` in pre stable versions is handled like `0.9.*` so active update is required and so I'm fine with this bc break and no additional flag.

File: UPGRADE.md [hunk 6]
```diff
@@ -1,5 +1,27 @@
...
+{
+    "formatUri": "/media/1/{format}/media-1.png?v=1-0"
+}
+```
+
+**After:**
+```json
+{
+    "formatUri": "/media/1/{format}/media-1.{extension}?v=1-0",
+    "preferredExtension": "png"
```
Comment (alexander-schranz) [inline UPGRADE.md]:
```suggestion
    "formatUri": "/media/1/{format}/media-1.{extension}?v=1-0",
    "formatPreferredExtension": "png"
```

**Comment by wachterjohannes** [comment]
> @alexander-schranz @wachterjohannes to prevent the BC break, we could maybe add a config like this:
> 
> ```yaml
> sulu_headless:
>     media:
>         use_preferred_extension: true
> ```
> 
> WDYT?

@Prokyonn would also go with a flag and throw a deprecation message when it is not set! else the PR LGTM.

**Comment by Prokyonn** [comment]
as discussed internally, we'll omit the configuration BC layer, because of
> if somebody is using ^0.9 they will not get automatically 0.10, as ^ in pre stable versions is handled like 0.9.* so active update is required and so I'm fine with this bc break and no additional flag.

**Review by alexander-schranz — APPROVED** [review]

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**